addons/sale_timesheet/models/account_move.py

```diff
@@ -41,47 +42,69 @@ def _compute_timesheet_count(self):
         for invoice in self:
             invoice.timesheet_count = mapped_data.get(invoice.id, 0)
 
+    def _has_timesheet_portal(self):
+        Timesheet = self.env['account.analytic.line']
+        initial_domain = Timesheet._timesheet_get_portal_domain()
+        domain = expression.AND([initial_domain, self.env['account.analytic.line']._timesheet_get_sale_domain(self.line_ids.sale_line_ids, self)])
+        return Timesheet.sudo().search(domain, limit=1)
+
     def action_view_timesheet(self):
         self.ensure_one()
-        return {
-            'type': 'ir.actions.act_window',
-            'name': _('Timesheets'),
-            'domain': [('project_id', '!=', False)],
-            'res_model': 'account.analytic.line',
-            'view_id': False,
-            'view_mode': 'list,form',
-            'help': _("""
-                <p class="o_view_nocontent_smiling_face">
-                    Record timesheets
-                </p><p>
-                    You can register and track your workings hours by project every
-                    day. Every time spent on a project will become a cost and can be re-invoiced to
-                    customers if required.
-                </p>
-            """),
-            'limit': 80,
-            'context': {
-                'default_project_id': self.id,
-                'search_default_project_id': [self.id]
-            }
-        }
-
-    def _link_timesheets_to_invoice(self, start_date=None, end_date=None):
+        action = self.env['ir.actions.actions']._for_xml_id('sale_timesheet.action_timesheet_from_invoice')
+        context = ast.literal_eval(action.get('context', '{}'))
+        context['default_is_so_line_edited'] = True
+
+        default_so_line = next((sale_line for sale_line in self.line_ids.sale_line_ids if sale_line.is_service and sale_line.product_id.service_policy != 'delivered_timesheet'), False)
```

Comment:
```suggestion
        default_so_line = next((
            sale_line for sale_line in self.line_ids.sale_line_ids
            if sale_line.is_service and sale_line.product_id.service_policy != 'delivered_timesheet'
        ), False)
```
? :slightly_smiling_face: